### DISCUSSION TITLE
prediction fail at metal bulk and slab

### DISCUSSION THREAD

**Discussion by yeol1389** [ACEsuit/mace#1076] [Q&A]
I am fine-tuning the mpa-0 model. While it performs well on organic molecules(), it fails to accurately predict Al and Li slab (surface) structures. I trained the model using 4×4×4-sized slabs, but it cannot even generalize to slightly larger 5×5×4 slabs. How should I construct the training set to improve this?

`mace_run_train \
    --name="MACE_VASP_FT" \
    --foundation_model="mace-mpa-0-medium.model" \
    --multiheads_finetuning=True \
    --train_file="$MERGED_EXTXYZ" \
    --valid_fraction=0.1 \
    --loss="weighted" \
    --energy_weight=1.0 \
    --forces_weight=1.0 \
    --E0s="{1: -1.1132324, 6: -1.2927434, 13: -0.24728363}" \
    --energy_key="energy" \
    --forces_key="forces" \
    --lr=0.01 \
    --lr_scheduler_gamma=0.95 \
    --scaling="rms_forces_scaling" \
    --batch_size=8 \
    --max_num_epochs=200 \
    --ema \
    --ema_decay=0.99 \
    --amsgrad \
    --default_dtype="float64" \
    --device=cuda \
    --seed=1 \
    --patience=20 \
    --restart_latest \
    --correlation=3 \`

this is about Al slab
<img width="589" height="589" alt="image" src="https://github.com/user-attachments/assets/f58986c8-9c33-4383-8894-81ee57149bd1" />
<img width="590" height="590" alt="image" src="https://github.com/user-attachments/assets/b41f38a3-4e2a-460a-807b-1a9127f5b1da" />

this is about organic molecules(There is still an energy offset, even for those predictions.)
<img width="434" height="435" alt="image" src="https://github.com/user-attachments/assets/4d74a1e7-1ba3-474a-a7f7-8c254c01f2af" />
<img width="434" height="435" alt="image" src="https://github.com/user-attachments/assets/c113e869-bf61-4498-ad81-6cbf665c8d42" />

**Comment by ilyes319** [comment]
Can you please post your log files and final error plots from training.

Envoyé à partir de Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________
From: yeol1389 ***@***.***>
Sent: Thursday, July 17, 2025 9:41:36 AM
To: ACEsuit/mace ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ACEsuit/mace] prediction fail at metal bulk and slab (Discussion #1076)


I am fine-tuning the mpa-0 model. While it performs well on organic molecules(), it fails to accurately predict Al and Li slab (surface) structures. I trained the model using 4×4×4-sized slabs, but it cannot even generalize to slightly larger 5×5×4 slabs. How should I construct the training set to improve this?

mace_run_train \ --name="MACE_VASP_FT" \ --foundation_model="mace-mpa-0-medium.model" \ --multiheads_finetuning=True \ --train_file="$MERGED_EXTXYZ" \ --valid_fraction=0.1 \ --loss="weighted" \ --energy_weight=1.0 \ --forces_weight=1.0 \ --E0s="{1: -1.1132324, 6: -1.2927434, 13: -0.24728363}" \ --energy_key="energy" \ --forces_key="forces" \ --lr=0.01 \ --lr_scheduler_gamma=0.95 \ --scaling="rms_forces_scaling" \ --batch_size=8 \ --max_num_epochs=200 \ --ema \ --ema_decay=0.99 \ --amsgrad \ --default_dtype="float64" \ --device=cuda \ --seed=1 \ --patience=20 \ --restart_latest \ --correlation=3 \

this is about Al slab
image.png (view on web)<https://github.com/user-attachments/assets/f58986c8-9c33-4383-8894-81ee57149bd1>
image.png (view on web)<https://github.com/user-attachments/assets/b41f38a3-4e2a-460a-807b-1a9127f5b1da>

this is about organic molecules(There is still an energy offset, even for those predictions.)
image.png (view on web)<https://github.com/user-attachments/assets/4d74a1e7-1ba3-474a-a7f7-8c254c01f2af>
image.png (view on web)<https://github.com/user-attachments/assets/c113e869-bf61-4498-ad81-6cbf665c8d42>

this is prediction about train data(Al 4x4x4)(Given that the predictions on the training data are accurate, it seems that the model has been trained well...)
image.png (view on web)<https://github.com/user-attachments/assets/7e49515e-d54e-40dc-b1d4-b4fc4b5bf682>
image.png (view on web)<https://github.com/user-attachments/assets/135106d6-c461-43d8-954a-490f83ec57b9>

**Reply by yeol1389** [reply]
thank you for fast response!
[MACE_VASP_FT_run-1.log](https://github.com/user-attachments/files/21286940/MACE_VASP_FT_run-1.log)

**Comment by gabor1** [comment]
Look at your energies/atom for your 4x4x4 and 5x5x4 slabs: the former is all around 4.4, the latter is 3.6. So your two slab calculations have vastly inconsistent energies. could it be that your k-point sampling is really bad?

**Reply by yeol1389** [reply]
When creating the training data with VASP, I enabled the D3 van der Waals correction. Could this be a wrong approach?

**Reply by gabor1** [reply]
I think it's better to generate training data without D3, fit to it, and you can add D3 to the MACE prediction afterwards (there are efficient D3 calculators now in lammps that use the gpu)

**Reply by vijaymocherla** [reply]
Just a follow up question related to this. Would it make sense to generate training data with one of the vdW functionals for calculations involving metal slabs. Also, this would be for a standalone MACE model and not for fine tunning the foundational models.
